### ISSUE TITLE
RoiTools.makeTiles: please add option for reverse order  

### ISSUE THREAD

**Issue by ofrag** [qupath/qupath#1277]
Hi,

We'd appreciate adding an option of reverse order to RoiTools.MakeTiles 
It would be very useful for us 

Thanks
Ofra

**Comment by petebankhead** [comment]
Hi Ofra,
Can you explain more the requirement please?
If you just want to reverse the returned list, in Groovy this would just involve calling `List.reverse()`

```groovy
def list = [1, 2, 3, 4] as List
def listReversed = list.reverse()
println list
println listReversed
```

**Comment by petebankhead** [comment]
How exactly are you making the annotations - from the user interface or in a script?

I'm a bit concerned about adding more complexity into QuPath for this, since that might in turn leads to yet more complexity (e.g. a requirement to create tiles from top to bottom....) and we're really trying to simplify. Therefore I'm not sure if changing the `makeTiles` method is the right approach to achieving the end goal.

**Comment by ofrag** [comment]
We are using the following script - adapted from a script on image.sc : 

`// File: tileAnnotation2.groovy

import qupath.lib.gui.QuPathGUI
import qupath.lib.gui.viewer.QuPathViewer
import qupath.lib.objects.PathAnnotationObject
import qupath.lib.objects.PathObject
import qupath.lib.objects.PathObjects
import qupath.lib.roi.interfaces.ROI
import qupath.lib.roi.RoiTools


// Adjust THIS (n : Number of Tiles)
int n = 10
def trimToROI = true
//parentClass = "IL-left"
childClass = "Tiles"

// Get the QuPath viewer and check if an object is currently selected
QuPathViewer viewer = QuPathGUI.getInstance().getViewer()
def objSelected = viewer.getSelectedObject()
parentClass  = objSelected.getPathClass()

// Calculate the width and height of the ROI
if (objSelected != null && objSelected instanceof PathAnnotationObject){
    ROI roi = ((PathAnnotationObject)objSelected).getROI()

    int w = roi.getBoundsWidth()
    int h = roi.getBoundsHeight()

    int tW, tH

// devide tiles based on the width 
        tW = Math.floor(w/n)+1
        tH = h

// Create a list of ROIs by using the makeTiles method from the RoiTools class, which divides the ROI into tiles based on the specified width and height.
    List<ROI> pathROIs = RoiTools.makeTiles(roi, tW, tH, trimToROI)

// Create a list of PathObject tiles and assigns a name to each tile
    List<PathObject> tiles = new ArrayList<>(pathROIs.size())

    Iterator<ROI> iter = pathROIs.iterator()
    int idx = 0
    while (iter.hasNext()) {
        try {
            PathObject tile = PathObjects.createAnnotationObject(iter.next(), getPathClass(childClass))           
            if (tile != null) {
                idx++
                tile.setName("Tile " + idx)
                tiles.add(tile)
            }
        } catch (InterruptedException e) {
            lastMessage = "Tile creation interrupted for " + objSelected
            return
        } catch (Exception e) {
            iter.remove()
        }
    }

// Add the tiles to the selected annotation object and update the viewer
    ((PathAnnotationObject)objSelected).addPathObjects(tiles);
    viewer.getImageData().getHierarchy().fireHierarchyChangedEvent(this, objSelected);
}

parentAnnotation = getAnnotationObjects().findAll{it.getPathClass() == getPathClass(parentClass)}[0]
childAnnotation = getAnnotationObjects().findAll{it.getPathClass() == getPathClass(childClass )}

for (it_Child in childAnnotation) { 
    getCurrentHierarchy().addPathObjectBelowParent(parentAnnotation, it_Child, true) }

**Comment by petebankhead** [comment]
What happens if you simply replace the `makeTiles` line with the following ?

```groovy
def pathROIs = RoiTools.makeTiles(roi, tW, tH, trimToROI).reversed()
```

**Comment by petebankhead** [comment]
@ofrag I read this thread again and today I understood :)

We've replaced `RoiTools.makeTiles` with a the new `Tiler` class that supports different alignment options. Hopefully it will make things much easier.
* https://github.com/qupath/qupath/pull/1350

I'll close this issue as the new class should be available soon when v0.5.0 is available.
